### PR TITLE
feat(frontend): add option to ignore course in progression checks

### PR DESCRIPTION
**Pull request by lhvy**

### CODE REVIEW

File: frontend/src/pages/ProgressionChecker/IgnoredCoursesSection/IgnoredCoursesSection.tsx
Comment (leonardo-fan) [inline frontend/src/pages/ProgressionChecker/IgnoredCoursesSection/IgnoredCoursesSection.tsx:1]: Hahaha I see what you were talking about before basically repeating `FreeElectiveSection.tsx`. If you want, it'd be useful to have a generic template that you could pass title/subheading1/subheading2 text into if we want to add more sections in the future/reduce the repetition.

File: frontend/src/pages/ProgressionChecker/IgnoredCoursesSection/IgnoredCoursesSection.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  return (
+    <Collapsible
+      title={
+        <Title level={1} id="Progression Ignored" className="text">
+          Progression Ignored
+        </Title>
+      }
+    >
+      <Title level={4} className="text">
+        You have {uoc} UOC worth of additional courses ignored from your progression
```
Comment (leonardo-fan) [inline frontend/src/pages/ProgressionChecker/IgnoredCoursesSection/IgnoredCoursesSection.tsx]: no need for "additional"

**Review by leonardo-fan — DISMISSED** [review]
Big props Lucas for taking the initiative for this new improvement/feature. Most things look good to me, I've noticed some fixes that would be good though:
- ignoring from progression doesn't get updated in the dashboard cards (refer below) -> add similar logic to the `storeUOC[group].curr` key
image
- can you please upload a new gif to the help menu so it's consistent with what we have now
image
- see comment about generic new progression checker section that you could feed elements/text into

File: frontend/src/pages/ProgressionChecker/IgnoredCoursesSection/IgnoredCoursesSection.tsx (hunk 1 above)
Comment (lhvy) [reply frontend/src/pages/ProgressionChecker/IgnoredCoursesSection/IgnoredCoursesSection.tsx]: Fixed

File: frontend/src/pages/ProgressionChecker/IgnoredCoursesSection/IgnoredCoursesSection.tsx
Comment (lhvy) [reply frontend/src/pages/ProgressionChecker/IgnoredCoursesSection/IgnoredCoursesSection.tsx:1]: Done 👍

**Review by leonardo-fan — APPROVED** [review]
perfecto lgtm :) thanks Lucas!

**Review by imagine-hussain — APPROVED** [review]
Looks good!!!!!!

**Pull request merged**